firefox-ios/Client/Frontend/Home/Homepage Rebuild/HomepageViewController.swift

```diff
@@ -608,12 +635,17 @@ final class HomepageViewController: UIViewController,
                 .FirefoxAccountChanged,
                 .TopSitesUpdated,
                 .DefaultSearchEngineUpdated:
-            store.dispatch(
-                TopSitesAction(
-                    windowUUID: self.windowUUID,
-                    actionType: TopSitesActionType.fetchTopSites
+            // To ensure that `numberOfTilesPerRow` is calculated on the main thread
+            mainQueue.ensureMainThread { [weak self] in
```

Comment:
this is because we are using `traitCollection` in that method, I also added a comment to that method as well describing this.